### PR TITLE
Add scIB vignette

### PR DESCRIPTION
**Pull request by lazappi**
This PR will add a vignette showing how to recreate the figures from the scIB paper

### CODE REVIEW

**Comment by lazappi** [comment]
The vignette now shows how to recreate the scIB summary figure. The result is pretty close but not exactly the same. The big difference is that the bars are coloured by score rather than rank, which makes them appear quite different.

Other small things I couldn't work out how to do (let me know if there is a solution):

- Align the column labels
- Move the column labels to the bottom
- Adjust the size/other styling of text columns/labels
- Add a palette to a text column
- Remove the dotted lines around bars

**Review by rcannood — APPROVED** [review]
Hey Luke! This PR is fantastic, thank you so much for creating it!

I created a few suggestions to make the code less wordy. Are these ok for you? This is the result of the vignette with suggestions included: https://rcannood.quarto.pub/recreating-the-scib-figures.

Regarding the remaining discrepancies:

* Colouring by rank: This is something that I'm working on allowing with the changes proposed in #9
* Legends don't show, add a palette to a text column: WIP in the refactor_legends branch
* Control over column label rendering: Should be tackled in a separate issue
* Remove the dotted lines around bars: Should also be tackled in a separate issue

Let me know if the proposed suggestions are okay with you.

File: vignettes/scIB.Rmd [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    label_simulations_2,
+    package_score,
+    package_label,
+    paper_score,
+    paper_label,
+    time_score,
+    time_label,
+    memory_score,
+    memory_label
+  ) |>
```
[the remark this answers is not in the document]
Comment (lazappi) [reply vignettes/scIB.Rmd:175]: Different coding style 😝. I wasn't sure what happens if there are extra columns so I think I was just being safe. If the extra columns aren't plotted makes sense to remove this.

File: vignettes/scIB.Rmd [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+We also describe the various column groups.
+
+```{r summary-col-groups}
+column_groups <- tibble::tribble(
+        ~ group,     ~ palette,      ~ level1,
+       "Method",      "Method",      "Method",
+          "RNA",         "RNA",         "RNA",
+  "Simulations", "Simulations", "Simulations",
+    "Usability",   "Usability",   "Usability",
+  "Scalability", "Scalability", "Scalability",
```
[the remark this answers is not in the document]
Comment (lazappi) [reply vignettes/scIB.Rmd:231]: Ah, I thought these would colour the main plot but just colouring the headings is much better. Thanks!

File: vignettes/scIB.Rmd [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+```{r summary-figure, fig.width=8, fig.height=8}
+funky_heatmap(
+  scib_summary_plot,
+  column_info = column_info,
+  column_groups = column_groups,
+  row_info = row_info,
+  palettes = palettes,
+  scale_column = FALSE,
+  col_annot_angle = 90,
+  col_annot_offset = 5
```
[the remark this answers is not in the document]
Comment (lazappi) [reply vignettes/scIB.Rmd:285]: Yeah, that's reasonable. I would prefer to have the vertical labels but I guess there's some stuff that needs to be implemented before that works properly.

**Comment by lazappi** [comment]
Thanks for the suggestions, I think those have improved things!

> * Control over column label rendering: Should be tackled in a separate issue
> * Remove the dotted lines around bars: Should also be tackled in a separate issue

Do you want me to open issues for these?

**Comment by rcannood** [comment]
> > * Control over column label rendering: Should be tackled in a separate issue
> > * Remove the dotted lines around bars: Should also be tackled in a separate issue
> 
> Do you want me to open issues for these?

Yes please!

**Review by rcannood — APPROVED** [review]

**Pull request merged**